airflow/api_fastapi/core_api/routes/ui/structure.py

```diff
@@ -63,4 +63,31 @@ def structure_data(
         "edges": edges,
     }
 
+    if external_dependencies:
+        entry_node_ref = nodes[0]
```

Comment:
Should we check for nodes or edges are empty?

Reply:
I'm not sure if a serialized dag can have no tasks. I added a test and apparently the dag is not saved to the db so I would say that `nodes` cannot be empty.

But to be safe I still added a check for that, thanks.